test/e2e_node/services/kubelet.go

```diff
@@ -302,6 +302,7 @@ func (e *E2EServices) startKubelet() (*server, error) {
 	}
 
 	cmd := exec.Command(cmdArgs[0], cmdArgs[1:]...)
+	restartOnExit := framework.TestContext.RestartKubelet
```

Comment:
can another variable be used to store the flag state instead of framework.TestContext.RestartKubelet ?

Reply:
Reading the flag from the file init is one option, I'm not sure if is a pattern we want to propagate here:

```
func init() {
	flag.Var(&kubeletArgs, "kubelet-flags", "Kubelet flags passed to kubelet, this will override default kubelet flags in the test. Flags specified in multiple kubelet-flags will be concatenate.")
	flag.BoolVar(&genKubeletConfigFile, "generate-kubelet-config-file", true, "The test runner will generate a Kubelet config file containing test defaults instead of passing default flags to the Kubelet.")
 	flag.BoolVar(&restartKubelet, "restart-kubelet", true, "If true, restart Kubelet unit when the process is killed.")

}
```

Reply:
It seems to be the recommended way to do `init()` here and use a local variable for the `restart-kubelet` flag based on `test/e2e/framework/test_context.go` ln#49-76?

Reply:
The point is to decouple specific node flags from the main e2e framework context. But adding these flags on the init() here without further context it's only augmenting the mess.

Reply:
@neolit123 we opened an issue to move this TestContext.RestartKubelet to a proper place, for this specific PR, we are keeping in the global one, wdyt?

Reply:
SGTM.